lib/pure/strutils.nim

```diff
@@ -2114,8 +2114,8 @@ func replace*(s, sub: string, by = "", maxOccurrence = -1): string {.rtl,
   ## * `replaceWord func<#replaceWord,string,string,string>`_
   ## * `multiReplace func<#multiReplace,string,varargs[]>`_
   runnableExamples:
-    doAssert "This is not {}, but can be used as {}".replace("{}", "fmt()") == "This is not fmt(), but can be used as fmt()"
-    doAssert "almost valid variable name".replace(" ", "_", 2) == "almost_valid_variable name"
+    assert "FooBarBaz".replace("Foo", "Bar") == "BarBarBaz"
```

Comment:
this doesn't illustrate that it replaces all occurrences since there's a single occurrence

Reply:
That is a stupid oversight, sorry